### DISCUSSION TITLE
FAN CONTROL, Lenovo ideapad 5 pro 16 7840hs APH8 780m 32GB

### DISCUSSION THREAD

**Discussion by Zetis659** [DavidS95/Smokeless_UMAF#46] [Q&A]
Hello everyone, please help me to solve the problem of fan noise when the laptop is idle. **The fans turn on even if the temperature is ±35°**. I would like to raise this threshold and make the laptop completely silent in office tasks, at temperatures up to 55-60°.

**That's how I tried to do it:**
I boot from a USB, and then I see a very fast inscription.
![smkls](https://github.com/DavidS95/Smokeless_UMAF/assets/96617763/041e9af6-634b-4c98-98fb-1170d52bca3c)
I'm not sure what it says about a successful download, but nevertheless it starts.

Then I go to **Device manager → AMD CBS → SMU common options**.
Here I set the Fan control values to manual mode and then set the temperatures and fan speeds I need.
![smkls2](https://github.com/DavidS95/Smokeless_UMAF/assets/96617763/37a5819d-bcb8-456e-9712-040930bc9040)

Then I save the changes by pressing F10 and then Y. After that, I go to the main menu, press the Continue button and press Enter again. When my OS (it doesn't matter if it's windows 11 or manjaro GNOME) boots, I check if the fan settings have been applied. No matter how I set them up, for some reason they are **NOT USED** and the fans continue to spin even at 35 °. **Please tell me how to solve this problem and apply the fan settings with the help of Smokeless_UMAF, because the bios (UEFI) of this laptop does not allow you to adjust the fans.**

**Comment by l33tm4st3r** [comment]
Do not press the F10 key, just change the settings to your preferences, then press the "Esc" key until a text box appears, press the "y" key, then go back to the main menu and press the "continue" option. 

This is how I do it on my laptop and it works perfectly. Remind you that if you change anything in your "normal" bios, you will have to repeat the process I have indicated.

**Reply by Zetis659** [reply]
Unfortunately, it didn't change anything. Fan settings are still not applied

**Comment by ghost** [comment]
The SMU is not able to control the fan on Legion and most other Laptop, that part is handled by the EC...

you might able to define a custom fan curve from the os, when in custom-mode

**Reply by Zetis659** [reply]
Thank you for your answer, but unfortunately neither in windows 10/11 nor in linux (I have manjaro GNOME), the OS does not show fan speed in any program. It's like he's not there. For example, windows shows only the temperature of the CPU, GPU, M2, TDP and the total power consumption of the system. Similarly in linux. If the system could see the fan speed, I think I could find a way to adjust them, but I still haven't found a way to see these fan speeds. Could you help me with this?)

**Reply by ghost** [reply]
Have you tried in vantage?
You need to be in custom power mode,(purple power led on legion)  for it to show

**Reply by Zetis659** [reply]
In lenovo vantage on my laptop there are only 3 settings modes: battery saving, intelligent cooling and performance. These profiles change only the CPU TDP: ≈30, ≈50, ≈60 watts, respectively, while they do not affect the fan speed in idle time or at very low loads such as watching videos. In any case, when the CPU reaches ≈38 degrees, the fan starts at a fairly noticeable speed, especially in a quiet room. The strangest thing is that the laptop in such loads as 2k/ 4k video does not reach temperatures even at 50 degrees, usually 40-45, but the fan still works for some reason, although in theory it could work perfectly without fans and not overheat at such temperatures. I would just like to raise its inclusion threshold to 50-55. So that when coding or watching videos, I could be in complete silence. Sometimes the fans do not work if the CPU temperature drops to 30-33 degrees, but this does not happen for a long time and in rather cold rooms. Unfortunately, I didn't find any custom power modes on my laptop that would solve my problem, but I think that's not even the point. Even when idle and consuming only 6-10 watts, the fans work 95% of the time and are relatively quiet, although they can still be heard and very rarely turn off completely. As soon as I start the videos they start up again.

**Comment by cad42s13** [comment]
I have an X13 with a 7840u and 32gb of ram.  I'd like to set dedicated VRAM to 16GB so I can run larger llama models.  These won't use shared gpu memory on windows without a lot of effort.

Would you please try going to Device Manager→AMD CBS→NBIO Common Option→GFX Configuration, then set "UMA Frame Buffer Size" to 16G and see if it works?  Windows Task Manager should show 16GB dedicated GPU memory on the task manager performance tab.

UMAF will not save when I change this setting.  It says "submit failed" and I am forced to discard the change before I can exit / continue.

I tried changing a couple of other settings and got the same results, but I was cautious and did not want to change too much.

**Reply by ghost** [reply]
It's a thinkpad, you cannot use standard umaf to change setting, these are locked up, on the ready to boot phase...
One way is to load a unsigned efi driver and use it as trampoline from the real bios.

TL;DR : Not possible with the public version of UMAF

**Reply by cad42s13** [reply]
Thanks. Had reached that conclusion from various threads. Will look into the suggestion, much appreciated.

**Reply by pmnlla** [reply]
Is there an unsigned EFI driver that you made we can load?

**Reply by cad42s13** [reply]
Thanks. I'm looking at other options that don't involve tweaking the bios. Very cool to know Umaf exists, and I'm sure it will be useful in future.

Thanks!

**Reply by leonardcoutinho** [reply]
> It's a thinkpad, you cannot use standard umaf to change setting, these are locked up, on the ready to boot phase... One way is to load a unsigned efi driver and use it as trampoline from the real bios.
> 
> TL;DR : Not possible with the public version of UMAF

Hi Ghost, you can make an version for thinkpad for us?

**Comment by ghost** [comment]
> Is there an unsigned EFI driver that you made we can load?

As long SecureBoot is off, you can load also unsiged driver

**Reply by pmnlla** [reply]
Secureboot is indeed off.  What driver would we use as a trampoline?

**Comment by TDuffinNTU** [comment]
Did you find an answer for this? I also noticed the fan curve options weren't being applied :( lm-sensors package on linux suggests fan control is done via ACPI but not sure where to go from there...

**Reply by Igradi** [reply]
any updates on this? I am trying to do the same for the ideapad 14aph8

**Comment by amongesimpasta** [comment]
Did you find a solution? I have a Ideapad 5 14ARE05. When saving the manual fan settings and rebooting they arent used.

**Comment by Blaubluter** [comment]
Bios to OLD

**Comment by PeterB2020** [comment]
Still no solution for changing Lenovo Ideapad fans?
UMAF is sadly not working and it's really frustrating that Lenovo is blocking the option...
